### ISSUE TITLE
Bug: git log changes-pattern parsing issue

### ISSUE THREAD

**Issue by KutsunaSubaRya** [kellyjonbrazil/jc#395]
Hi!
Firstly, thank's for the great tools.
I've noticed that regex of `changes-pattern` in `git_log.py` and `git_log_s.py` may cause problems in no insertion situation.

part of git log
```
commit 35e0e9c32afbc5450c661bcb3a23904d135358ad
Author:     Kelly Brazil <kellyjonbrazil@gmail.com>
AuthorDate: Tue Mar 22 15:57:43 2022 -0700
Commit:     Kelly Brazil <kellyjonbrazil@gmail.com>
CommitDate: Tue Mar 22 15:57:43 2022 -0700

    remove print statement

 jc/parsers/asciitable.py | 1 -
 1 file changed, 1 deletion(-)
```

after parse by `jc --git-log`
```json
{
    "commit":"35e0e9c32afbc5450c661bcb3a23904d135358ad",
    "author":"Kelly Brazil",
    "author_email":"kellyjonbrazil@gmail.com",
    "date":"Tue Mar 22 15:57:43 2022 -0700",
    "commit_by":"Kelly Brazil",
    "commit_by_email":"kellyjonbrazil@gmail.com",
    "commit_by_date":"Tue Mar 22 15:57:43 2022 -0700",
    "stats":
    {
        "files_changed":1,
        "insertions":1,
        "deletions":0,
        "files":["jc/parsers/asciitable.py"]
    },
    "message":"remove print statement",
    "epoch":1647989863,
    "epoch_utc":null
}
```

It seems that the groups of insertions and deletions are not the optional parts of pattern.

The regex below functions correctly.
```python
changes_pattern = re.compile(r'\s(?P<files>\d+)\s+(files? changed)(?:,\s+(?P<insertions>\d+)\s+(insertions?\(\+\)))?(?:,\s+(?P<deletions>\d+)\s+(deletions?\(\-\)))?')
```

**Comment by kellyjonbrazil** [comment]
Hi - thank you for reporting this and providing a fix! I'll test this fix and get it into the next release.

Thanks again!

**Comment by kellyjonbrazil** [comment]
Sorry I didn't notice your PR and I had already updated the code on my side along with the tests. This is now up on the `dev` branch and will be included in the next release. Thank you very much for the fix!

**Comment by kellyjonbrazil** [comment]
Released in v1.23.2
